tools/codegen/api/lazy.py

```diff
@@ -89,45 +88,70 @@ def isWrappedScalarType(typ: Type) -> bool:
         return typ.name == BaseTy.Scalar
     elif isinstance(typ, (OptionalType, ListType)):
         return isWrappedScalarType(typ.elem)
-    else:
-        return False
+    return False
+
+def isGeneratorType(typ: Type) -> bool:
+    if isinstance(typ, BaseType):
+        return typ.name == BaseTy.Generator
+    elif isinstance(typ, (OptionalType)):
+        return isGeneratorType(typ.elem)
+    return False
+
+class LazyArgument:
+    name: str
+    orig_type: Type
+    lazy_type: CType
+    is_wrapped_scalar: bool
+    is_generator: bool
+
+    # true if this argument is or contains a lazy IR value
+    is_lazy_value: bool
+
+    def __init__(self, arg: Argument):
+        self.name = arg.name
+        self.orig_type = arg.type
+        self.is_generator = isGeneratorType(arg.type)
+        if self.is_generator:
+            self.lazy_type = None
+        else:
+            self.lazy_type = process_ir_type(arg.type)
+        self.is_wrapped_scalar = isWrappedScalarType(arg.type)
 
+        self.is_lazy_value = isValueType(self.lazy_type)
 
 # Inspired by a FunctionSchema object, a LazyIrSchema holds the schema of a Lazy IR node.
 # Unlike a FunctionSchema, it has no round-trippable string form (relating to the YAML),
 # but carries type information from a native FunctionSchema modified for use with IR nodes,
 # and preserving original argument names.
-
-
 class LazyIrSchema:
     # The name of the operator this function schema describes.
     name: 'OperatorName'
 
-    positional_arg_types: Tuple[NamedCType, ...]
-    keyword_arg_types: Tuple[NamedCType, ...]
+    positional_args: Tuple[LazyArgument, ...]
+    keyword_args: Tuple[LazyArgument, ...]
```

Comment:
oh yeah, you mentioned in a previous PR that we actually do need to care about the difference between positional and keyword args, and I just realized, I guess that's because the TorchScript IR actually distinguishes them ([code](https://github.com/pytorch/pytorch/blob/2e79740fcf2c40dfe4166d660e8f72ad700fc70f/lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/ts_node_lowering.cpp#L438))